### PR TITLE
8260309: Shenandoah: Clean up ShenandoahBarrierSet

### PR DESCRIPTION
**Pull request by rkennke**
We collected some cruft in ShenandoahBarrierSet. Time to clean it up. 

This fixes/removes a number of includes, fixes some comments and it also removes is_a() and is_aligned() which look like leftovers/requirements from earlier incarnations of the superclass BarrierSet. Using the override keyword would be useful for such situations (btw, are we ok to start using override, nullptr, auto etc in Shenandoah, or do we want to keep it C++ for backporting ease?)

One thing I was not sure about is the ShenandoahHeap* _heap field. Making it const will likely help the compiler avoid repeated access (e.g. in a number of perf-critical paths like the LRB impl). However, maybe we should get rid of the field altogether and make it explicitely using ShenandoahHeap::heap() and avoid repeated access instead of helping the compiler and hoping for the best?

Testing:
 - [x] hotspot_gc_shenandoah release, fastdebug

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8260309](https://bugs.openjdk.java.net/browse/JDK-8260309): Shenandoah: Clean up ShenandoahBarrierSet

### Reviewers
 * [Aleksey Shipilev](https://openjdk.java.net/census#shade) (@shipilev - **Reviewer**)
 * [Zhengyu Gu](https://openjdk.java.net/census#zgu) (@zhengyu123 - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2202/head:pull/2202`
`$ git checkout pull/2202`

### CODE REVIEW

**Comment by shipilev** [comment]
What is up with this PR? There are no recorded changes, and tests are failing ;)

**Comment by rkennke** [comment]
> What is up with this PR? There are no recorded changes, and tests are failing ;)

Weird, I seem to have messed it up. I'll fix it shortly.

**Comment by rkennke** [comment]
> > What is up with this PR? There are no recorded changes, and tests are failing ;)
> 
> Weird, I seem to have messed it up. I'll fix it shortly.

It should be good to review, now.

File: src/hotspot/share/gc/shenandoah/shenandoahBarrierSet.inline.hpp [hunk 1]
```diff
@@ -25,7 +25,7 @@
 #ifndef SHARE_GC_SHENANDOAH_SHENANDOAHBARRIERSET_INLINE_HPP
 #define SHARE_GC_SHENANDOAH_SHENANDOAHBARRIERSET_INLINE_HPP
 
-#include "gc/shared/barrierSet.hpp"
+#include "gc/shared/accessBarrierSupport.hpp"
```
Comment (shipilev) [inline src/hotspot/share/gc/shenandoah/shenandoahBarrierSet.inline.hpp:28]: Should it be `accessBarrierSupport.inline.hpp`? Other `*BarrierSet.inline.hpp`-s seem to include that.

**Review by shipilev — APPROVED** [review]
Looks fine, but I have a minor question.

File: src/hotspot/share/gc/shenandoah/shenandoahBarrierSet.inline.hpp (hunk 1 above)
Comment (rkennke) [reply src/hotspot/share/gc/shenandoah/shenandoahBarrierSet.inline.hpp]: Right. I changed that. Thanks!

**Review by shipilev — APPROVED** [review]

**Review by zhengyu123 — APPROVED** [review]
Looks good.

**Pull request closed without merging**